packages/sitecore-jss/src/personalization/layout-fragment-service.ts

```diff
@@ -0,0 +1,95 @@
+import { ComponentRendering } from '../layout/models';
+import { GraphQLRequestClient } from './../graphql-request-client';
+import debug from './../debug';
+
+export interface LayoutFragmentData {
+  fragment: ComponentRendering;
+}
+
+export interface LayoutFragmentService {
+  fetchLayoutFragmentData(
+    routePath: string,
+    language: string,
+    renderingId: string,
+    ruleId: string
+  ): Promise<LayoutFragmentData>;
+}
+
+export type GraphQLLayoutFragmentServiceConfig = {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
```

Comment:
Is siteName required?
For example, Sitecore site can be resolved via a host name.

Reply:
SiteName is required, it is one of the query arg to resolve the fragment on ExpEdge